### ISSUE TITLE
long type parameter cause `Expected a single space`

### ISSUE THREAD

**Issue by david-claimer** [pinterest/ktlint#2635]
## Expected Behavior
```kotlin
    data class TestClass(
        val someTypeParam:
            Map<List<Map<Pair<String, String>, Pair<String, String>>>, List<Map<Pair<String, String>, Pair<String, String>>>>,
        val b: String,
    )
```
running `./gradlew :ktlintJvmMainSourceSetCheck` should not produce any error as far as I am aware

## Observed Behavior
```
`Expected a single space`
```

## Steps to Reproduce
Put that class in a file and run `./gradlew :ktlintJvmMainSourceSetCheck`

## Your Environment

* Version of ktlint used: 1.0.1
* Relevant parts of the `.editorconfig` settings
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): org.jlleitschuh.gradle.ktlint 12.1.0
* Version of Gradle used (if applicable): 7.4
* Operating System and version: OSX

**Comment by paul-dingemans** [comment]
This problem seems to be resolved in Ktlint `1.1.1`:

```
$ ktlint-1.0.1 --stdin
11:48:28.780 [main] INFO com.pinterest.ktlint.cli.internal.KtlintCommandLine - Enable default patterns [**/*.kt, **/*.kts]
data class TestClass(
    val someTypeParam:
        Map<List<Map<Pair<String, String>, Pair<String, String>>>, List<Map<Pair<String, String>, Pair<String, String>>>>,
    val b: String,
)
<stdin>:2:23: Expected a single space (standard:parameter-list-spacing)
<stdin>:3:1: Exceeded max line length (100) (standard:max-line-length)

Summary error count (descending) by rule:
  standard:max-line-length: 1
  standard:parameter-list-spacing: 1

$ ktlint-1.1.1 --stdin
11:48:45.395 [main] INFO com.pinterest.ktlint.cli.internal.KtlintCommandLine -- Enable default patterns [**/*.kt, **/*.kts]
data class TestClass(
    val someTypeParam:
        Map<List<Map<Pair<String, String>, Pair<String, String>>>, List<Map<Pair<String, String>, Pair<String, String>>>>,
    val b: String,
)
<stdin>:3:101: Exceeded max line length (100) (standard:max-line-length)

Summary error count (descending) by rule:
  standard:max-line-length: 1
```

Please upgrade to latest Ktlint `1.2.1`.
